packages/@webex/plugin-cc/src/services/task/index.ts

```diff
@@ -86,5 +109,144 @@ export default class Task implements ITask {
     }
   }
 
-  // TODO: Hold/resume, recording pause/resume, consult and transfer public methods to be implemented here
+  /**
+   * This is used to hold the task.
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.hold(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   * */
+  public async hold(): Promise<TaskResponse> {
+    try {
+      if (!this.data.mediaResourceId || this.data.mediaResourceId.length === 0) {
+        throw new Error('MediaResourceId is required');
+      }
+
+      return this.contact.hold({
+        interactionId: this.data.interactionId,
+        data: {mediaResourceId: this.data.mediaResourceId},
+      });
+    } catch (error) {
+      const {error: detailedError} = getErrorDetails(error, 'hold', CC_FILE);
+      throw detailedError;
+    }
+  }
+
+  /**
+   * This is used to resume the task.
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.resume(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async resume(): Promise<TaskResponse> {
+    try {
+      if (!this.data.mediaResourceId || this.data.mediaResourceId.length === 0) {
```

Comment:
If we know that empty mediaresourceId fails the hold/resume, would we still need this condition. Isn't contact layer handling this?

Reply:
This is a valid point... the more I see it, I think we can forego this entire condition... it anyways will fail for an empty mediaResourceId.
Is that alright @Kesari3008?

Reply:
yes that's alright